<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0 minimum-scale=1.0 maximum-scale=1.0">
    <meta name="robots" content="noindex,nofollow">
    <meta name="googlebot" content="noindex,nofollow">
    <title>에러 발생 확인하기</title>
    <style>
        body{
            margin: 1rem;
        }
        h2{
            display: flex;
            align-items: center;
            margin: 2rem 0 10px;
        }
        h2:before{
            content: '';
            width: 7px; 
            height: 7px;
            margin: -3px 7px 0 0;
            background: rgba(125, 0, 255, 1);
        }
        h3{ 
            margin-top: 3rem;
            padding-bottom: 7px;
            border-bottom: 1px solid rgba(80, 0, 255, 0.2);
        }
        
        table{
            width: 100%;
            border-collapse: collapse;
            margin-bottom: 1rem;
        }
        thead{
            background: rgba(125, 0, 255, 0.05);
        }
        th, td {
            padding: 10px;
            border: 1px solid rgba(125, 0, 255, 0.1);
        }

        section{
            margin: 2rem 0 0;
        }

        .rel{
            margin-top: 5rem;
        }
    </style>
</head>
<body>

    <h1>에러 발생 확인하기</h1>

    <h2>Syntax</h2>

    <table>
        <thead>
            <th>구문</th>
            <th>의미</th>
        </thead>
        <tbody>
            <tr>
                <th>try {} catch (error) {}</th>
                <td>에러 처리</td> 
            </tr>
        </tbody>
    </table>

    <p>
        에러가 발생하면 이후의 작업은 중단되지만 작업 내용에 따라 에러가 발생해도 처리를 지속해야 할 경우 
        try와 catch를 사용한다. 
    </p>

    <script>
        const a = 10

        try {
            a = 20
        } catch (error) {
            console.log(`에러가 발생했습니다. ${error.message}`)
        }

        // 중단 없이 실행
        console.log(`상수 a의 값은 ${a}입니다.`)
    </script>


    <p class="rel">출처 : 자바스크립트 코드 레시피 278</p>

</body>
</html>